### ISSUE TITLE
local https for serve-web

### ISSUE THREAD

**Issue by crbyxwpzfl** [microsoft/vscode#203730]
#### tldr
with `code serve-web --host 0.0.0.0  --without-connection-token` Jupyter notebooks alias `.ipynb` files won’t work and stay blank see blow 

![screenshot](https://github.com/microsoft/vscode/assets/60987359/a2582258-3067-4f53-a08e-7f758ce57b1b)

![screencap](https://github.com/microsoft/vscode/assets/60987359/6edcba43-e5ec-409c-9f73-85cdd5dc1452)

to resolve this is use [caddy](https://coder.com/docs/code-server/latest/guide#using-lets-encrypt-with-caddy) with `sudo caddy run —config /etc/caddy/Caddyfile`
```sh
pi.local {
        reverse_proxy 127.0.0.1:8000
}
```
to put a reverse proxy with local signed tls cert in front of it like @connor4312 suggested [here](https://github.com/microsoft/vscode/issues/191276) and removed `—host 0.0.0.0` as this defaults to localhost only

#### more info
this is probably the wrong place to put this but the issue mentioned above is locked i want to document this for me and perhaps others. other than that the code cli features are so cool and simple to use. furthermore I know this very insecure but since I use this locally only with a pi connected via usbc to a ipad this seems tolerable to me please let me know otherwise. anyways please close this if this is irrelevant or covered else where.

#### other findings
with just `code serve-web --host 0.0.0.0` or with `—connection-token`  im not able to connect via a browser on a ipad to the pi. perhaps the browser strips the token duno

**Comment by connor4312** [comment]
I don't have any plans to do this; the `serve-web` interface is not designed as a public-facing server at the moment. Using Caddy or something similar is the way to go.
